packages/core/src/database/index.ts

```diff
@@ -215,22 +232,64 @@ export const createDatabase = ({
         ? [preBuild.databaseConfig.poolConfig.max - internalMax, 0, 0]
         : [equalMax, equalMax, equalMax];
 
+    const internal = createPool({
+      ...preBuild.databaseConfig.poolConfig,
+      application_name: `${preBuild.namespace}_internal`,
+      max: internalMax,
+      statement_timeout: 10 * 60 * 1000, // 10 minutes to accommodate slow sync store migrations.
+    });
+
+    const connection = (parse as unknown as typeof parse.parse)(
+      preBuild.databaseConfig.poolConfig.connectionString!,
+    );
+
+    const role = `ponder_readonly_${connection.database}_${preBuild.namespace}`;
+
+    await internal.query(`CREATE SCHEMA IF NOT EXISTS "${preBuild.namespace}"`);
+    const hasRole = await internal
+      .query("SELECT FROM pg_roles WHERE rolname = $1", [role])
+      .then(({ rows }) => rows[0]);
+    if (hasRole) {
+      await internal.query(`DROP OWNED BY ${role}`);
+      await internal.query(`DROP ROLE IF EXISTS ${role}`);
+    }
+    await internal.query(
+      `CREATE ROLE ${role} WITH LOGIN PASSWORD '${connection.password}'`,
```

Comment:
We should probably use a different password than the one from the main/admin/superuser connection string. Looks like PG requires a non-null/non-empty password, so we could use something like "public" or "readonly", or generate a less predictable one. cc @jaylmiller

Reply:
default to something predictable but make option to override would be good